File: CoWorkHub/CoWorkHub.Model/WorkspaceType.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class WorkspaceType
    {
        public int WorkspaceTypeId { get; set; }
        public string TypeName { get; set; } = null!;
        public bool IsDeleted { get; set; } = false;
    }
}

File: CoWorkHub/UI/coworkhub_desktop/lib/models/workspace_type.dart
import 'package:json_annotation/json_annotation.dart';

part 'workspace_type.g.dart';

@JsonSerializable()
class WorkspaceType {
  int workspaceTypeId;
  String typeName;
  bool isDeleted;

  WorkspaceType({
    required this.workspaceTypeId,
    required this.typeName,
    this.isDeleted = false,
  });

  factory WorkspaceType.fromJson(Map<String, dynamic> json) =>
      _$WorkspaceTypeFromJson(json);
  Map<String, dynamic> toJson() => _$WorkspaceTypeToJson(this);
}
